### PR TITLE
Add stats for custom scoring feature

### PR DESCRIPTION
**Pull request by jmazanec15**
*Issue #, if available:*

*Description of changes:*
Adds a few statistics for custom scoring feature. The following statistics are added:
#### script_compilations
The number of times the knn script is compiled. This value should only be 0 or 1 most of the time. However, if the cache containing the compiled scripts is filled, it may cause the script to be recompiled.

#### script_compilation_errors
The number of errors during script compilation.

#### script_query_requests
The number of query requests that use the k-NN score script.

#### script_query_errors
The number of errors that have occurred during use of the k-NN score script.

Along with adding the stats, I updated the README and added another stats integration test to verify the changes.

I think in the future, it will probably make sense to refactor the stats api into different sections for custom scoring and approximate k Nearest Neighbor search. I will create an issue to do this.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/test/java/com/amazon/opendistroforelasticsearch/knn/plugin/action/RestKNNStatsHandlerIT.java [hunk 1]
```diff
@@ -164,6 +170,62 @@ public void testInvalidNodeIdStats() throws Exception {
         assertEquals(0, nodeStats.size());
     }
 
+    /**
+     *  Test checks that script stats are properly updated
+     */
+    public void testScriptStats() throws Exception {
```
Comment (vamshin) [inline src/test/java/com/amazon/opendistroforelasticsearch/knn/plugin/action/RestKNNStatsHandlerIT.java]: Can we add similar test with multiple shards?

File: README.md [hunk 2]
```diff
@@ -352,6 +352,18 @@ The total time in nanoseconds it has taken to load items into cache (cumulative)
...
+#### script_compilations
+The number of times the knn script is compiled. This value should only be 0 or 1 most of the time. However, if the cache containing the compiled scripts is filled, it may cause the script to be recompiled.
+
+#### script_compilation_errors
+The number of errors during script compilation.
+
+#### script_query_requests
+The number of query requests that use the k-NN score script.
+
+#### script_query_errors
```
Comment (vamshin) [inline README.md:364]: I think these errors are counted at shard level. We could update the documentation accordingly. Also you might want to check if all the above metrics at shard level or index level

Comment (jmazanec15) [reply README.md:364]: Yes, they are at shard level. I updated README

File: src/test/java/com/amazon/opendistroforelasticsearch/knn/plugin/action/RestKNNStatsHandlerIT.java (hunk 1 above)
Comment (jmazanec15) [reply src/test/java/com/amazon/opendistroforelasticsearch/knn/plugin/action/RestKNNStatsHandlerIT.java]: Sure

**Review by vamshin — APPROVED** [review]
LGTM! Thanks for the stats.

**Pull request merged**